src/transformers/trainer.py

```diff
@@ -1690,8 +1690,8 @@ def _save_checkpoint(self, model, trial, metrics=None):
                 xm.save(self.lr_scheduler.state_dict(), os.path.join(output_dir, SCHEDULER_NAME))
                 reissue_pt_warnings(caught_warnings)
         elif is_sagemaker_mp_enabled():
-            if smp.dp_rank() == 0:
-                # Consolidate the state dict on all processed of dp_rank 0
+            if smp.rdp_rank() == 0:
```

Comment:
Shouldn't this test whether `smp.state.cfg.prescaled_batch` is `True` or not like in #15811 ?

Reply:
Not necessarily, prescaled batch only affects how the data is read. Parameters on a rank is independent of that. For checkpointing, all we care about is the parameters, so we should use rdp rank in all cases.